### PR TITLE
Linear Algebra Topic Update

### PR DESCRIPTION
**Pull request by mmujtabah**
Added a matrix reshape topic in linear algebra and related problem.

### CODE REVIEW

File: app/lin_alg_problems.py [hunk 1]
```diff
@@ -145,7 +145,43 @@
...
         {"test": "transpose_matrix([[1,2,3],[4,5,6]])", "expected_output": "[[1, 4], [2, 5], [3, 6]]"}
     ]
-}, "Calculate Mean by Row or Column (easy)":{
+}, "Reshape Matrix (easy)": {
+    "description": "Write a Python function that reshapes a given matrix into a specified shape.",
+    "example": """Example:
+    input: a = [[1,2,3,4],[5,6,7,8]], new_shape = (4, 2)
+    output: [[1, 2], [3, 4], [5, 6], [7, 8]]
+    reasoning: The given matrix is reshaped from 2x4 to 4x2.""",
+    "video": "https://youtu.be/DNoLs5tTGAw?si=vpkPobZMA8YY10WY",
```
Comment (moe18) [inline app/lin_alg_problems.py:154]: Change the link from the url to "coming soon"

**Review by moe18 — CHANGES_REQUESTED** [review]
Looks really good, added a small comment

File: app/lin_alg_problems.py (hunk 1 above)
Comment (mmujtabah) [reply app/lin_alg_problems.py:154]: Updated the URL.

**Review by mmujtabah — COMMENTED** [review]
Requested changes have been made.

**Pull request merged**